Remove BITCOIN_CASH ifdef completely

sickpig opened the pull request:
Since release 1.2.0.0 BUCash release enforce the use of the new cash net magic bit, e.g. incoming connections that used the legacy magic are reject now, this PR removes: 

- NODE_BITCOIN_CASH service bit and related code 
- `fUsesCashMagic` logic.

The former was used to implement preferential peering toward cash nodes in the period of time  when we share the same magic bit along with bitcoin legacy node. 
The latter was used to avoid banning cash nodes that instantiate incoming connections using the legacy net magic bit.

The PR is marked as WIP cause I need to refactor `buip055.py`, the test is temporarely disabled.

~(***This PR depends on #939***)~

sickpig committed:
Remove NODE_BITCOIN_CASH service bit

Since we are enforcing the use of the cash net magic bit both for
incoming and outgoing connections the cash service bit is not useful
anymore.

Temporarely disabled `buip055.py` test.

sickpig commented:
travis failure due to formatting issue on `src/init.cpp` 

https://travis-ci.org/BitcoinUnlimited/BitcoinUnlimited/jobs/327890830#L706

sickpig committed:
Remove fUsesCashMagic logic

There's no need to have a mechanism to handle to different net magic
with the same codebase. Hence we don't need to determine which set to
use at runtime anymore.

sickpig commented:
rebased on top of #918

sickpig committed:
Remove newdaaactivationtime from allowed args list

This should have been removed in 11ff236 ("Activate the new DAA by
height") but it slipped through.

sickpig commented:
rebased

sickpig commented:
rebased

sickpig committed:
Always use BCH as ticker in estimatesmartfee RPC call

sickpig committed:
Remove BITCOIN_CASH_FORK_HEIGHT from unlimited.h enum

Use uahfHeight consensus parameter instead.

sickpig committed:
Remove last instance fo BITCOIN_CASH from unlimited.h

Slightly reword comments around SINGLE_PEER_REQUEST_MODE_AGE enum value
and add a TODO point about the set value itself.

sickpig committed:
Support only CASH nodes seeders (mainnet and testnet)

sickpig committed:
Remove all BITCOIN_CASH ifdefs from Qt related code

sickpig committed:
Always enforce LOW_S and NULLFAIL if 13th Nov '17 HF is active

This is a leftover from commit 70dba3cf9499e9f13f6aea67d0faa07fb811cca7
that should have removed all ifdefs around Nov 13th 2017 HF (new DAA).

sickpig committed:
Last pass in removing BITCOIN_CASH ifdef

Last bits were in cpp unit tests, wallet and scripts interpreter code.

sickpig commented:
- remove all instances of `BITCOIN_CASH` `ifdef`
- rebased on top of current `dev`
- rabased on top of #939

sickpig commented:
Taking out of WIP, I will take care of `buip055.py` refactoring on a separate PR.

ptschip commented on src/init.cpp:
```diff
@@ -820,12 +820,7 @@ bool AppInit2(Config &config, boost::thread_group &threadGroup, CScheduler &sche
-#endif
+    // BUIP010 Xtreme Thinblocks: end section
 
     nMaxTipAge = GetArg("-maxtipage", DEFAULT_MAX_TIP_AGE);
```
I don't think we should be removing the bitcoin cash service bit just yet...i agree this needs to be sunsetted but first we need to remove only when BU nodes disconnect from nodes that do not present the bit.  Later , perhaps in a year, we should reclaim the bit.  This is what XT is doing and I believe is correct since there may be other nodes (older nodes for one that are expecting the bit) and posssibly other wallets or services that are still expecting the bit to be present.

sickpig replied:
agreed. CASH service bit reinstated.

ptschip commented on src/net.cpp:
```diff
@@ -1777,20 +1773,6 @@ void ThreadOpenConnections()
-                    nDisconnects++;
-                }
-            }
-#endif
```
This should be removed , yes, but we should still present the service bit in case older nodes or other services still are expecting it.

ptschip replied:
I would recommend making this a separate PR and merge right away for the bug fix release...this disconnecting of nodes doesn't need to be there anymore...

sickpig committed:
Reinstate CASH service bit.

As noticed by ptschip there are still implementations that requires CASH
service bit as pre-condition to establish outgoing connections
(see: https://github.com/bitcoinxt/bitcoinxt/issues/309)

This revert ade8a0f modulo the BITCOIN_CASH ifdef and the
min-bitcoin-cash-nodes cash logic.

sickpig commented:
rebased and add the bitcoin cash service bit back modulo the BITCOIN_CASH ifdef and the `-min-bitcoin-cash-nodes` cash logic (as per @ptschip observation)

ptschip reviewed (approved):
tested and works

gandrewstone commented on src/rpc/misc.cpp:
```diff
@@ -116,8 +116,9 @@ UniValue getinfo(const UniValue &params, bool fHelp)
     if (cashFork)
-        cashFork = cashFork->GetAncestor(BITCOIN_CASH_FORK_HEIGHT);
+        cashFork = cashFork->GetAncestor(consensusParams.uahfHeight);
     std::string fork = "Bitcoin";
```
the code from line 121 to 130 writes "bitcoin" if you are pre-fork and "bitcoin cash" if you are post fork.  But at this point we are post fork -- I don't think that watching it switch is meaningful during IBD.  In fact its probably going to cause bugs because people will use this "fork" field to verify what client flavor is being used.

so we can delete 119 to 129 and replace 130 with: obj.push_back(Pair("fork", "Bitcoin Cash"));

sickpig replied:
makes sense

sickpig committed:
Modify getinfo rpc to always return "Bitcoin Cash" as current fork.

The rpc calls returned "Bitcoin" if you were at a pre-UAHF fork height and
"Bitcoin Cash" if you were post fork. But at this point we are post fork
hence watching the fork field switch is meaningless during IBD. In
fact its probably going to cause bugs report because people will use this
"fork" field to verify what client flavor is being used.
